### DISCUSSION TITLE
Apply materials on Maya special duplicate instances

### DISCUSSION THREAD

**Discussion by raphaeljnwave** [Autodesk/maya-usd#1954] [Q&A] [answered]
I try to convert a maya scene into a USD file. In my scene, I have objects that are instances (duplicate special). On these instances is apply the same material.

When I export the scene using the mayaUsd plugin, I see that my instances are converted to Xform with instanceable=true and references to an over prim in the same file (/MayaExportedInstanceSources/myGroup/myMesh).

But it seems there is no way to link a material to 'myMesh', I guess that is why material on instances are not managed by default. But How can I do it by keeping all the data inside a single usd file ? Am I obliged to export my instances with their material into another usd file and to reference that other usd file from my main one ?

Thank you very much,
Raphael

**Comment by JGamache-autodesk** [comment, marked as answer]
It seems to all be in choosing the right locations for the materials. MayaUSD does not do that correctly at the moment.

Materials in instance prototypes can not be declared outside the prototype or at the root of the prototype. This means it needs to appear at minimum one level down in the prototype.

This complicates things and will require reworking the material export code to do a better job of finding where to put these materials and even duplicate them if they are shared with primitives outside of the prototype.

I have attached a very primitive scene (I'm a dev...) that showcases materials on instances. The default is a red coat, but the last two were turned to blue using [Collection-based](https://graphics.pixar.com/usd/release/wp_usdshade.html) and regular assignment overrides. These overrides do not seem to be able to peek inside instances, as I can not find a way to only make the arms blue. That would probably require asking for support on the USD-interest mailing list.

Can you please open an issue so we can track the missing instanced materials?
[Humans.zip](https://github.com/Autodesk/maya-usd/files/7929339/Humans.zip)

**Reply by raphaeljnwave** [reply]
Thank you @JGamache-autodesk for your explanation and usd example !
Indeed, when I was testing, it seemed that we can override the reference materials with only one single material (in your case BlueSG). Is that what you are saying ?
Sure, if I well understood your point, I can create an issue.

Thank you

**Reply by JGamache-autodesk** [reply]
We can apply single materials at strategic locations in the instance and if marked with "strongerThanDescendants" they will override the material declared in the instance. This is how I got a blue material on the last 2 instances. What I really would like is to be able to peek inside the instance and change materials there using collection-based assignment.

If you open the attached [ArmsOverride.zip](https://github.com/Autodesk/maya-usd/files/7934411/ArmsOverride.zip) in usdview and navigate to "/locator3/Torso/LArm" in the tree view, you will notice that "Resolved Full Material" is indeed BlueSG, but this information does not show in the viewport.
